torch/serialization.py

```diff
@@ -479,8 +498,10 @@ def persistent_load(saved_id):
         else:
             raise RuntimeError("Unknown saved id type: %s" % saved_id[0])
 
+    f_can_tell_and_seek = _can_tell_and_seek(f)
     f_should_read_directly = _should_read_directly(f)
-    if f_should_read_directly and f.tell() == 0:
+
+    if f_should_read_directly and f_can_tell_and_seek and f.tell() == 0:
```

Comment:
You don't need to use `f_can_tell_and_seek` in the conditional -- if f can't tell/seek, we've thrown an error to the user so so this line won't be called